presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralEncoder.java

```diff
@@ -257,9 +258,38 @@ else if (type instanceof RowType) {
             return rowType.getTypeParameters().stream()
                     .allMatch(LiteralEncoder::isSupportedLiteralType);
         }
+        else if (type instanceof MapType) {
+            MapType mapType = (MapType) type;
+            return isSupportedLiteralType(mapType.getKeyType()) && isSupportedLiteralType(mapType.getValueType());
+        }
         return SUPPORTED_LITERAL_TYPES.contains(type.getJavaType());
     }
 
+    public static long estimatedSize(Object object)
```

Comment:
I like the `SizeInBytes` notation better but seems like there are already plenty of `Size` without units so no strong preference.